CMakeLists.txt

```diff
@@ -0,0 +1,260 @@
+  jeeps/gpsrqst.cc jeeps/gpssend.cc jeeps/gpsserial.cc jeeps/jgpsutil.cc
+  jeeps/gpsusbread.cc jeeps/gpsusbsend.cc
+  jeeps/gpsusbcommon.cc
+)
+
+set(SUPPORT
+  queue.cc route.cc waypt.cc filter_vecs.cc util.cc vecs.cc mkshort.cc
+  csv_util.cc strptime.c grtcirc.cc util_crc.cc xmlgeneric.cc
+  formspec.cc xmltag.cc cet.cc cet_util.cc fatal.cc rgbcolors.cc
+  inifile.cc garmin_fs.cc gbsleep.cc units.cc gbser.cc
+  gbfile.cc parse.cc session.cc main.cc globals.cc
+  src/core/usasciicodec.cc
+  src/core/xmlstreamwriter.cc 
+)
+
+set(HEADERS
+  an1sym.h
+  cet.h
+  cet/ansi_x3_4_1968.h
+  cet/cp1252.h
+  cet/iso_8859_8.h
+  cet_util.h
+  csv_util.h
+  defs.h
+  explorist_ini.h
+  filter.h
+  filterdefs.h
+  garmin_device_xml.h
+  garmin_fs.h
+  garmin_gpi.h
+  garmin_tables.h
+  gbfile.h
+  gbser.h
+  gbser_private.h
+  gbversion.h
+  grtcirc.h
+  heightgrid.h
+  holux.h
+  inifile.h
+  jeeps/garminusb.h
+  jeeps/gps.h
+  jeeps/gpsapp.h
+  jeeps/gpscom.h
+  jeeps/gpsdatum.h
+  jeeps/gpsdevice.h
+  jeeps/gpsfmt.h
+  jeeps/gpsinput.h
+  jeeps/gpsmath.h
+  jeeps/gpsmem.h
+  jeeps/gpsport.h
+  jeeps/gpsproj.h
+  jeeps/gpsprot.h
+  jeeps/gpsread.h
+  jeeps/gpsrqst.h
+  jeeps/gpssend.h
+  jeeps/gpsserial.h
+  jeeps/gpsusbcommon.h
+  jeeps/gpsusbint.h
+  jeeps/gpsutil.h
+  magellan.h
+  mapsend.h
+  navilink.h
+  queue.h
+  session.h
+  shapelib/shapefil.h
+  strptime.h
+  xmlgeneric.h
+  zlib/crc32.h
+  zlib/deflate.h
+  zlib/gzguts.h
+  zlib/inffast.h
+  zlib/inffixed.h
+  zlib/inflate.h
+  zlib/inftrees.h
+  zlib/trees.h
+  zlib/zconf.h
+  zlib/zlib.h
+  zlib/zutil.h
+  src/core/datetime.h
+  src/core/file.h
+  src/core/logging.h
+  src/core/usasciicodec.h
+  src/core/xmlstreamwriter.h
+  src/core/xmltag.h
+)
+
+string(REPLACE .cc .h FILTER_HEADERS "${FILTERS}")
+set(HEADERS ${HEADERS} ${FILTER_HEADERS})
+
+include_directories(AFTER zlib)
+
+INCLUDE (CheckIncludeFile)
+if(UNIX)
+  # this is used by zlib
+  CHECK_INCLUDE_FILE("unistd.h" HAVE_UNISTD_H)
+  if(${HAVE_UNISTD_H})
+    add_definitions(-DHAVE_UNISTD_H)
+  endif()
+  # this is used by zlib
+  CHECK_INCLUDE_FILE("stdarg.h" HAVE_STDARG_H)
+  if(${HAVE_STDARG_H})
+    add_definitions(-DHAVE_STDARG_H)
+  endif()
+  add_definitions(-DHAVE_NANOSLEEP -DHAVE_LIBUSB -DHAVE_GLOB)
+  set(SOURCES ${SOURCES} gbser_posix.cc)
+  set(HEADERS ${HEADERS} gbser_posix.h)
+  set(JEEPS ${JEEPS} jeeps/gpslibusb.cc)
+  include_directories(AFTER jeeps)
+  add_compile_options(-O2 -Wall)
+endif()
+
+if(WIN32)
+  add_definitions(-D__WIN32__ -D_CONSOLE)
+  remove_definitions(-DUNICODE -DZLIB_INHIBITED)
+  if(CMAKE_BUILD_TYPE STREQUAL Debug)
+    add_definitions(-D_DEBUG)
+  endif()
+  set(SOURCES ${SOURCES} gbser_win.cc)
+  set(HEADERS ${HEADERS} gbser_win.h)
+  set(JEEPS ${JEEPS} jeeps/gpsusbwin.cc)
+  set(LIBS ${LIBS} setupapi)
+endif()
+
+if(MSVC)
+  add_definitions(-D_CRT_SECURE_NO_DEPRECATE)
+  add_compile_options(/MP -wd4100)
+endif()
+
+if (UNIX AND NOT APPLE)
+  add_definitions(-DHAVE_LINUX_HID)
```

Comment:
With delbin safely behind us, I don't think anything uses HID. I'd hope nothing does in the future - it's a terrible choice for machine protocols.